### PR TITLE
GitHub Actions CI

### PR DESCRIPTION
**Pull request by NattyNarwhal**
This:

* Builds the driver on all supported PHP versions for Windows (7.4, 8.0, 8.1)
* Builds the driver on all supported PHP versions for Ubuntu (7.3, 7.4, 8.0, 8.1), starts Db2 for Linux in a Docker container, and runs the test suite on it
* Makes any changes needed to make the test suite pass

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+jobs:
+  # XXX: macOS, Windows
+  ubuntu:
+    strategy:
+      matrix:
+          version: ['7.3', '7.4', '8.0', '8.1']
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install DB2 library
+        # XXX: Can this be optimized to not constantly redownload? Or be put in a better place?
```
Comment (kadler) [inline .github/workflows/ci.yml:12]: https://github.com/actions/cache

Comment (kadler) [reply .github/workflows/ci.yml:12]: Definitely would be helpful to cache, I think. I know the PRs for python-ibmdb were failing for a while due to flakiness in the IBM download site.

**Comment by NattyNarwhal** [comment]
Beaten into better shape, though the PHP 8.0 failure is bizarre; it's as if it was trying to link it as a normal executable instead of a library?

**Comment by kadler** [comment]
> Beaten into better shape, though the PHP 8.0 failure is bizarre; it's as if it was trying to link it as a normal executable instead of a library?

`make V=1`?

**Comment by kadler** [comment]
Seems to be some something weird with libtool. It's calling `libtool --mode=compile cc` which seems to be not passing the `-c` flag.

File: php_pdo_ibm.h [hunk 2]
```diff
@@ -47,12 +47,12 @@ PHP_MINFO_FUNCTION(pdo_ibm);
 	Declare any global variables you may need between the BEGIN
 	and END macros here: 
 */
-ZEND_BEGIN_MODULE_GLOBALS(pdo_ibm)
 #ifdef PASE /* i5/OS ease of use turn off/on */
+ZEND_BEGIN_MODULE_GLOBALS(pdo_ibm)
```
[the remark this answers is not in the document]
Comment (NattyNarwhal) [reply php_pdo_ibm.h:51]: Not a bad idea, but considering ibm_db2 went a long while without any non-PASE globals, might also be needless, yeah.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+          Hostname=localhost
+          Protocol=TCPIP
+          Port=60000
+          Database=sample
+          EOF
+          mkdir database
+          docker pull ibmcom/db2
+          docker run --name db2 --privileged=true -p 60000:50000 -e DB2INST1_PASSWORD=password -e LICENSE=accept -e DBNAME=sample -v database:/database -itd ibmcom/db2
+          docker ps -as
+          sleep 120 # XXX: NOT CORRECT, block on the docker run instead, somehow
```
Comment (kadler) [inline .github/workflows/ci.yml:58]: FYI, this is what I did for ibm_db: https://github.com/ibmdb/python-ibmdb/blob/master/scripts/setup-docker.sh#L12

**Pull request merged**